pets/etcd/etcd-petset.yaml

```diff
@@ -0,0 +1,121 @@
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+  name: etcd
+  labels:
+    app: etcd
+spec:
+  ports:
+  - port: 2380
+    name: peer
+  - port: 2379
+    name: client
+  clusterIP: None
+  selector:
+    app: etcd
+---
+apiVersion: apps/v1alpha1
+kind: PetSet
+metadata:
+  name: etcd
+spec:
+  serviceName: "etcd"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: etcd
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      containers:
+      - name: etcd
+        image: gcr.io/google_containers/etcd-amd64:2.2.5
+        ports:
+        - containerPort: 2380
+          name: peer
+        - containerPort: 2379
+          name: client
+        env:
+        - name: CLUSTER_SIZE
+          value: "3"
+        - name: PETSET_NAME
+          value: "etcd"
+        volumeMounts:
+        - name: datadir
+          mountPath: /var/run/etcd
+        command:
+          - "/bin/sh"
+          - "-ec"
+          - |
+            for i in $(seq 0 $((${CLUSTER_SIZE} - 1))); do
+                while true; do
+                    echo "Waiting for ${PETSET_NAME}-${i}.${PETSET_NAME} to come up"
+                    ping -W 1 -c 1 ${PETSET_NAME}-${i}.${PETSET_NAME} > /dev/null && break
```

Comment:
do you need to wait for all peers or can you bootstrap the first one in standalone mode, then join them one by one? the disadvantage of doing this is:

1. that you can't specify a readiness probe (depending on the probe, it would probably deadlock here since you won't pass it till you get past this, but the petset won't scale till the pet is in ready). 

2. Scaling is harder

If you want to simply wait till you are up (and get a list of all previous peers), this simple script might help: https://github.com/kubernetes/contrib/tree/master/pets/peer-finder, it just does a dns resolution and sorts the list. You can pass your bash as --on-start or --on-change